src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -4488,7 +4502,30 @@ internal SafeContext GetValEscape(BoundExpression expr, SafeContext localScopeDe
                     return GetValEscape(((BoundNullCoalescingAssignmentOperator)expr).RightOperand, localScopeDepth);
 
                 case BoundKind.IncrementOperator:
-                    return GetValEscape(((BoundIncrementOperator)expr).Operand, localScopeDepth);
+                    var increment = (BoundIncrementOperator)expr;
+                    if (increment.MethodOpt is { IsStatic: true } incrementMethod)
+                    {
+                        var prefix = increment.OperatorKind is UnaryOperatorKind.UserDefinedPrefixIncrement or UnaryOperatorKind.UserDefinedPrefixDecrement;
+                        Debug.Assert(prefix || increment.OperatorKind is UnaryOperatorKind.UserDefinedPostfixIncrement or UnaryOperatorKind.UserDefinedPostfixDecrement);
+
+                        // Prefix increment can be analyzed like the underlying method call since that's what it returns.
+                        // Postfix increment is better analyzed as only the operand since that's what it returns.
+                        if (prefix)
+                        {
+                            return GetInvocationEscapeScope(
```

Comment:
>GetInvocationEscapeScope [](http://example.com/codeflow?start=35&length=24)

Is this code path covered by a test? I.e. is there a test that is going to fail if this line is commented out?

Reply:
Adding `RefEscapingTests.UserDefinedIncrementOperator_RefStruct_Scoped_Prefix_Arg`.